modules/editor-service-endpoint/src/main/java/org/opencastproject/editor/endpoint/EditorRestEndpoint.java

```diff
@@ -0,0 +1,197 @@
+          pathParameters = { @RestParameter(name = "mediapackageid", description = "The id of the media package",
+                  isRequired = true, type = RestParameter.Type.STRING) }, responses = {
+          @RestResponse(description = "Media package found", responseCode = SC_OK),
+          @RestResponse(description = "Media package not found", responseCode = SC_NOT_FOUND) })
+  public Response getEditorData(@PathParam("mediapackageid") final String mediaPackageId) {
+    try {
+      EditingData response = editorService.getEditData(mediaPackageId);
+      if (response != null) {
+        return Response.ok(response.toString(), MediaType.APPLICATION_JSON_TYPE).build();
+      } else {
+        logger.error("EditorService returned empty response");
+        return RestUtil.R.serverError();
+      }
+    } catch (EditorServiceException e) {
+      switch(e.getErrorStatus()) {
+        case MEDIAPACKAGE_NOT_FOUND:
+          return RestUtil.R.notFound(String.format("Event '{%s}' not Found", mediaPackageId),
+                  MediaType.TEXT_PLAIN_TYPE);
+        case WORKFLOW_ACTIVE:
+          return RestUtil.R.locked();
+        case NO_INTERNAL_PUBLICATION:
+          return RestUtil.R.badRequest(e.getMessage());
+        case UNABLE_TO_CREATE_CATALOG:
+        case WORKFLOW_ERROR:
+        case WORKFLOW_NOT_FOUND:
+        case UNKNOWN:
+        default:
+          return RestUtil.R.serverError();
+      }
+    }
+  }
+
+  @POST
+  @Path("{mediapackageid}/edit.json")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "editVideo", description = "Takes editing information from the client side and processes it",
+          returnDescription = "",
+          pathParameters = {
+          @RestParameter(name = "mediapackageid", description = "The id of the media package", isRequired = true,
+                  type = RestParameter.Type.STRING) },
+          responses = {
+          @RestResponse(description = "Editing information saved and processed", responseCode = SC_OK),
+          @RestResponse(description = "Media package not found", responseCode = SC_NOT_FOUND),
+          @RestResponse(description = "The editing information cannot be parsed",
+                  responseCode = HttpServletResponse.SC_BAD_REQUEST) })
+  public Response editVideo(@PathParam("mediapackageid") final String mediaPackageId,
+          @Context HttpServletRequest request) {
+    String details;
+    try (InputStream is = request.getInputStream()) {
+      details = IOUtils.toString(is, request.getCharacterEncoding());
+    } catch (IOException e) {
+      logger.error("Error reading request body:", e);
+      return RestUtil.R.serverError();
+    }
+    logger.info("Editor POST-Request received: {}", details);
+
+    JSONParser parser = new JSONParser();
+    EditingData editingInfo;
+    try {
+      editingInfo = EditingData.parse(details);
+    } catch (Exception e) {
+      logger.warn("Unable to parse concat information ({})", details, e);
```

Comment:
We shouldn't warn admins if users do something wrong (e.g. ask for a non-existing identifier). This will just spam the logs. Let's drop this to debug. You will return an error anyway.
```suggestion
      logger.debug("Unable to parse concat information ({})", details, e);
```

Reply:
done